### PR TITLE
Do not allow both click selection and checkbox selection in same data…

### PR DESCRIPTION
**Pull request by jeff-phillips-18**
… list

### CODE REVIEW

File: test/views/datalist/data-list.spec.js [hunk 1]
```diff
@@ -179,6 +182,10 @@ describe('Directive:  pfDataList', function () {
       return item.uuid === '2';
     };
 
+    // allow item selection
+    $scope.listConfig.selectItems = true;
+    $scope.listConfig.showSelectBox = false;
+
     $scope.listConfig.checkDisabled = checkDisabled;
```
Comment (dtaylor113) [inline test/views/datalist/data-list.spec.js]: Would it be good to test an error case and that the exception is thrown by specifing:
$scope.listConfig.selectItems = true;
$scope.listConfig.showSelectBox = true;

**Comment by dtaylor113** [comment]
Couldn't a click selection just check the checkbox if shown?

File: src/views/datalist/data-list-directive.js [hunk 2]
```diff
@@ -214,12 +214,22 @@ angular.module('patternfly.views').directive('pfDataList', [
...
         }
       ],
 
       link: function (scope, element, attrs) {
         attrs.$observe('config', function () {
           scope.config = $.extend(true, angular.copy(scope.defaultConfig), scope.config);
+          if (scope.config.selectItems && scope.config.showSelectBox) {
+            throw new Error('pfDataList - ' +
+            'Illegal use of pfDataList directive! ' +
+            'Cannot allow both select box and click selection in the same data list.');
```
Comment (waldenraines) [inline src/views/datalist/data-list-directive.js]: These two occurrences of throwing this error should be DRYed.

**Comment by waldenraines** [comment]
Nitpick but could you reword the commit message so that it's < 50 characters as it's a [best practice] (https://github.com/blog/926-shiny-new-commit-styles).

**Comment by jeff-phillips-18** [comment]
@dtaylor113 no, the checkboxes are used when the click event has different purpose. Typically for navigation/drill down.

File: src/views/datalist/data-list-directive.js (hunk 2 above)
Comment (jeff-phillips-18) [inline src/views/datalist/data-list-directive.js]: Actually, I need to remove this second occurrence. There is no way to catch the error so it's not really helping to have it thrown.

**Comment by dtaylor113** [comment]
Ok, LGTM

File: test/views/datalist/data-list.spec.js (hunk 1 above)
Comment (jeff-phillips-18) [inline test/views/datalist/data-list.spec.js]: Added the test

**Comment by jeff-phillips-18** [comment]
@waldenraines Updated the commit message

**Comment by waldenraines** [comment]
@jeff-phillips-18 thanks!

**Pull request merged**